service/resource/certificate/desired.go

```diff
@@ -30,28 +30,36 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 
 	r.logger.Log("debug", "fetching certificates")
 	certificateFiles := []certificateFile{}
+
 	for _, service := range validServices {
-		namespace, name := prometheus.GetCertificateName(service)
+		clusterID := prometheus.GetClusterID(service)
 
-		certificate, err := r.k8sClient.CoreV1().Secrets(namespace).Get(name, metav1.GetOptions{})
-		if errors.IsNotFound(err) {
-			return nil, microerror.Maskf(missingError, "certificate %s/%s", namespace, name)
-		} else if err != nil {
-			return nil, microerror.Maskf(err, "an error occurred fetching certificate %s/%s", namespace, name)
+		certificates, err := r.k8sClient.CoreV1().Secrets(r.certificateNamespace).List(metav1.ListOptions{
+			LabelSelector: fmt.Sprintf(
+				"clusterComponent=%s, clusterID=%s",
+				r.certificateComponentName,
+				clusterID,
+			),
+		})
+		if err != nil {
+			return nil, microerror.Maskf(err, "an error occurred fetching certificate for cluster: %s", clusterID)
 		}
 
-		groupName := prometheus.GetGroupName(service)
+		if len(certificates.Items) == 0 {
+			return nil, microerror.Maskf(missingError, "certificate for cluster: %s", clusterID)
+		}
+		certificate := certificates.Items[0]
```

Comment:
Why do we only use the first item? What if there are two? I feel like this should be part of the error handling as well.

Reply:
For now, I know there's only one certificate. The certificatetpr sorts this in future.